File: problems/1337-the-k-weakest-rows-in-a-matrix/solution.java
import java.util.PriorityQueue;

class Solution {
    public int[] kWeakestRows(int[][] mat, int k) {
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> {
            if (a[0] != b[0]) {
                return a[0] - b[0];
            } else {
                return a[1] - b[1];
            }
        });

        for (int i = 0; i < mat.length; i++) {
            int soldiers = 0;
            for (int j = 0; j < mat[i].length; j++) {
                if (mat[i][j] == 1) {
                    soldiers++;
                } else {
                    break;
                }
            }
            pq.offer(new int[]{soldiers, i});
        }

        int[] result = new int[k];
        for (int i = 0; i < k; i++) {
            result[i] = pq.poll()[1];
        }

        return result;
    }
}

File: problems/1337-the-k-weakest-rows-in-a-matrix/solution.py
class Solution:
    def kWeakestRows(self, mat: List[List[int]], k: int) -> List[int]:
        soldiers = []
        for i, row in enumerate(mat):
            soldiers.append((sum(row), i))
        
        soldiers.sort()
        
        result = []
        for i in range(k):
            result.append(soldiers[i][1])
        
        return result